Library/Homebrew/utils.rb

```diff
@@ -770,3 +770,34 @@ def number_readable(number)
   (numstr.size - 3).step(1, -3) { |i| numstr.insert(i, ",") }
   numstr
 end
+
+# True if this version of Ruby supports text encodings in its strings
+def ruby_has_encoding?
+  String.method_defined?(:force_encoding)
+end
+
+# Truncates a text string to fit within a byte size constraint,
+# preserving character encoding validity. The returned string will
+# be not much longer than the specified max_bytes, though the exact
+# shortfall or overrun may vary.
+def truncate_text_to_approximate_size(s, max_bytes, options = {})
+  front_weight = options.fetch(:front_weight, 0.5)
+  return s if s.bytesize <= max_bytes
+
+  if front_weight < 0.0 || front_weight > 1.0
```

Comment:
I think it's worth moving this range check further up to where `front_weight` is fetched, so that option fetching and validation are in the same spot and the validation happens every time, not just when the input happens to exceed the given limit. This will also help avoid the scenario where this function is invoked incorrectly, but isn't noticed because most inputs tend to be below the threshold. (The only reason I can think of to have the validation here is to elide the cost of the check for smaller inputs, but this sounds like premature optimization to me.)